vita3k/renderer/src/scene.cpp

```diff
@@ -223,44 +223,93 @@ COMMAND(handle_transfer_copy) {
     const SceGxmTransferType src_type = helper.pop<SceGxmTransferType>();
     const SceGxmTransferType dst_type = helper.pop<SceGxmTransferType>();
 
-    if (src_type == dst_type) {
-        const auto src_bpp = gxm::get_bits_per_pixel(src->format);
-        const auto dest_bpp = gxm::get_bits_per_pixel(dest->format);
-        const uint32_t src_bytes_per_pixel = (src_bpp + 7) >> 3;
-        const uint32_t dest_bytes_per_pixel = (dest_bpp + 7) >> 3;
+    // Get bits per pixel of source and destination
+    const auto src_bpp = gxm::get_bits_per_pixel(src->format);
+    const auto dest_bpp = gxm::get_bits_per_pixel(dest->format);
+
+    // Set bytes per pixel of source and destination
+    const uint32_t src_bytes_per_pixel = (src_bpp + 7) >> 3;
+    const uint32_t dest_bytes_per_pixel = (dest_bpp + 7) >> 3;
 
-        for (uint32_t x = 0; x < src->width; x++) {
-            for (uint32_t y = 0; y < src->height; y++) {
+    // Set width and height of source and destination
+    const uint32_t src_width = src->width;
+    const uint32_t src_height = src->height;
+    const uint32_t dest_width = dest->width;
+    const uint32_t dest_height = dest->height;
+
+    const auto copy_pixel = [&](const uint32_t src_offset, const uint32_t dest_offset) {
+        // Set pointer of source and destination
+        const auto src_ptr = (uint8_t *)src->address.get(mem) + src_offset;
+        auto dest_ptr = (uint8_t *)dest->address.get(mem) + dest_offset;
```

Comment:
It would be better to get the pointer to src and dest (the src->adress.get(mem) and dest->address.get(mem)) at the beginning of the function and not inside the loop<;

Reply:
? no understund for that, is point and offset, no have sense set point alone